File: javascript/raindrops/raindrops.js
module.exports = function() {
  this.convert = function(int) {
    var result = '';

    if (int % 3 === 0) {
      result += 'Pling';
    }
    if (int % 5 === 0) {
      result += 'Plang';
    }
    if (int % 7 === 0) {
      result += 'Plong';
    }

    return result || '' + int;
  };
};

File: java/raindrops/src/main/java/Raindrops.java
public class Raindrops {
  public static String convert(int num) {
    StringBuilder builder = new StringBuilder();

    if (num % 3 == 0) {
      builder.append("Pling");
    }
    if (num % 5 == 0) {
      builder.append("Plang");
    }
    if (num % 7 == 0) {
      builder.append("Plong");
    }
    String res = builder.toString();

    if (res.length() == 0) return Integer.toString(num);
    return res;
  }
}
